lib/Db/MessageMapper.php

```diff
@@ -508,22 +508,26 @@ public function deleteByUid(Mailbox $mailbox, int ...$uids): void {
 			->from($this->getTableName())
 			->where(
 				$messageIdQuery->expr()->eq('mailbox_id', $deleteRecipientsQuery->createNamedParameter($mailbox->getId())),
-				$messageIdQuery->expr()->in('uid', $deleteRecipientsQuery->createNamedParameter($uids, IQueryBuilder::PARAM_INT_ARRAY))
+				$messageIdQuery->expr()->in('uid', $deleteRecipientsQuery->createParameter('uids'))
 			);
-		// delete all related recipient entries
-		$deleteRecipientsQuery->delete('mail_recipients')
-			->where($deleteRecipientsQuery->expr()->in('message_id', $messageIdQuery->createFunction($messageIdQuery->getSQL()), IQueryBuilder::PARAM_INT_ARRAY));
-		$deleteRecipientsQuery->execute();
+		foreach (array_chunk($uids, 1000) as $chunk) {
+			$messageIdQuery->setParameter('uids', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
```

Comment:
Is going to conflict with: https://github.com/nextcloud/mail/pull/4932

Reply:
4932 is merged